### ISSUE TITLE
[Feature]: [CPU][GLM5Next] Add native sparse MLA / KeyPool indexer support for GLM-5.3-Flash

### ISSUE THREAD

**Issue by ixcans** [vllm-project/vllm#57345]
## Motivation

GLM-5.3-Flash (`Glm5NextForConditionalGeneration` / `glm5next`) is now supported in vLLM, but its native sparse MLA / KeyPool attention path is not implemented for the CPU backend.

This is different from #54018 / #54029.

Those address the older `glm_moe_dsa` path and provide a way to disable sparse attention and fall back to dense MLA on CPU.

For GLM-5.3-Flash, I would like to request **native CPU execution of its sparse MLA + KeyPool/KeyPoolIndexer path**, preserving the model's sparse-attention behavior rather than converting it to dense MLA.

## Current state on `main`

The generic CPU attention selector still rejects sparse attention:

```python
if attn_selector_config.use_sparse:
    raise NotImplementedError("Sparse Attention is not supported on CPU.")
```

At the same time, CPU support has recently progressed significantly: DeepSeek-V4 now has its own x86/AMX CPU sparse-MLA/indexer implementation, including CPU kernels for the sparse attention/indexing pipeline.

`vllm/platforms/cpu.py` explicitly documents the DeepSeek-V4 CPU kernels under:

```text
csrc/cpu/sgl-kernels/
├── flash_mla
├── store_cache
├── compressor
├── paged_mqa_logits
└── topk
```

and configures DeepSeek-V4 separately with its required block size.

This makes GLM5Next support seem like a useful next target rather than requiring CPU sparse attention infrastructure to be created entirely from scratch.

## GLM5Next-specific gap

GLM-5.3 uses a different sparse-indexing scheme from DeepSeek-V4.

The current GLM5Next sparse-indexer dispatcher is effectively:

```python
if current_platform.is_rocm():
    from .amd.sparse_indexer import SparseAttnIndexerKpool
else:
    from .nvidia.sparse_indexer import SparseAttnIndexerKpool
```

There is no CPU branch.

Therefore a CPU build falls into the NVIDIA implementation rather than a CPU implementation.

GLM5Next's attention implementation also currently imports:

```python
from vllm.models.glm5next.nvidia.ops.kpool_compress import fwht128_quant_fp8
```

and constructs the model-specific:

```text
SparseAttnIndexerKpool
Glm5NextIndexerCache
Glm5NextTailCache
KeyPool/KPool compression
top-k sparse selection
```

The GLM5Next indexer has additional semantics that are not identical to the DeepSeek-V4 Lightning Indexer:

* NoPE MLA
* KeyPool/KPool compression
* per-token KeyPool gate score
* FP8 indexer K cache
* tail cache for the current/incomplete pool
* top-k pool/token selection
* KeyPool state that must remain synchronized with the MLA KV cache

So simply routing GLM5Next through the existing generic CPU MLA backend would not be sufficient.

## Requested feature

Add a native CPU implementation of the GLM5Next sparse MLA / KeyPool pipeline.

A possible architecture could be:

```text
GLM5Next hidden states
        │
        ├── indexer Q/K projections
        │
        ├── KeyPool compression
        │
        ├── KeyPool / tail-cache update
        │
        ├── indexer score computation
        │
        ├── top-k sparse selection
        │
        ▼
selected MLA KV entries
        │
        ▼
CPU / AMX MLA
```

It may be possible to reuse parts of the recently added DeepSeek-V4 CPU sparse-MLA implementation, especially:

* paged/cache addressing
* top-k
* MQA score computation
* sparse metadata
* AMX MLA kernels
* chunked-prefill / prefix-cache integration

while implementing the GLM5Next-specific KeyPool compression and cache semantics separately.

## Possible implementation direction

I am not proposing a specific internal API, but one possible approach would be:

1. Add an explicit CPU branch in `vllm/models/glm5next/sparse_indexer.py`.

```python
if current_platform.is_rocm():
    ...
elif current_platform.is_cpu():
    from .cpu.sparse_indexer import SparseAttnIndexerKpool
else:
    ...
```

2. Implement a CPU/reference version of the GLM5Next KPool indexer first.

3. Add optimized x86 paths where useful for:

   * FWHT + quantization
   * KPool compression
   * paged MQA logits
   * top-k selection
   * MLA decode

4. Reuse the existing AMX MLA backend where the GLM5Next NoPE geometry permits it.

5. Support both:

   * chunked prefill
   * decode

6. Preserve:

   * prefix caching
   * hybrid KV-cache grouping
   * `k_cache`
   * `tail_cache`

A correctness-first implementation would already be valuable even if the initial indexer is slower than the NVIDIA/AMD paths. AMX/AVX-512 optimization could follow.

## Why dense fallback is not sufficient

Falling back to dense MLA is useful for compatibility and CI, as discussed in #54018/#54029, but it removes the primary computational benefit of GLM-5.3's sparse attention at long context lengths.

For CPU inference in particular, avoiding unnecessary KV-memory traffic is important, so native sparse execution is much more valuable than making the model merely boot with dense attention.

## Suggested acceptance criteria

* GLM-5.3-Flash can construct its MLA/indexer layers on `VLLM_TARGET_DEVICE=cpu`.
* CPU does not import/use NVIDIA-only KPool operators.
* Positive `index_topk` remains enabled.
* Sparse KeyPool selection is actually performed rather than silently using dense MLA.
* Prefill works.
* Decode works.
* Chunked prefill works.
* Prefix caching works.
* KeyPool `k_cache` and `tail_cache` remain synchronized correctly.
* Results match a GPU reference within expected BF16/FP8 tolerance.
* x86 AMX is used where supported/appropriate.

## Relationship to existing CPU work

This request is **not** a generic "please add sparse attention to CPU" issue.

There is already model-specific DeepSeek-V4 CPU sparse-attention/indexer support on current `main`.

The request is to extend CPU sparse-attention support to **GLM5Next's KeyPool architecture**, ideally reusing/commonizing the new DeepSeek-V4 CPU infrastructure where practical.

It is also separate from #54018/#54029, which concern `glm_moe_dsa` and dense fallback.

Would it make sense to build the GLM5Next CPU KeyPool implementation on top of the recently added DeepSeek-V4 CPU sparse-MLA/indexer infrastructure, or should GLM5Next initially have a separate CPU backend because its KeyPool semantics differ?

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by kunkunblueberry** [comment]
I opened draft PR #57687 for a correctness-first CPU sparse MLA / KeyPool path for GLM5Next: https://github.com/vllm-project/vllm/pull/57687. The implementation keeps the common model graph, adds GLM-local CPU indexer/MLA adapters, and includes CPU correctness/runtime tests. Full model generation and native optimization remain follow-up work.
